### ISSUE TITLE
[Question] Attibutes in custom timeseries widget

### ISSUE THREAD

**Issue by NicoWii4** [thingsboard/thingsboard#6193]
**Component**
 
  * Generic
 
 **Description**
 Hello,
 I created a custom timeserie widget using "widgets library" menu.
 I want to use both timeseries inputs and attributes inputs. Unfortunately, attributes inputs are not allowed.
 There is an old issue that describe the same problem I'm facing:  [https://github.com/thingsboard/thingsboard/issues/1737](https://github.com/thingsboard/thingsboard/issues/1737).
 For now, I use attribute service in my widget JS code but it is not adaptive and it causes loading problems in my dashboard sometimes.

Is there any available solution to  allow attributes inputs in my custom widget ?

Thank you for your answer.

 **Environment**
 
  * OS:  Ubuntu 20.04
  * ThingsBoard: 3.3.3PE
  * Browser: Mozilla Firefox 97.0

**Comment by ashvayka** [comment]
We are updating time-series widgets(data-source) to support attributes. This feature will be available in release 3.4.

**Comment by ChrisSchMA** [comment]
@ashvayka I am currently running Thingsboard 3.4 and I cannot seem to find a way to add attributes to my time-series widgets. Am I missing something, or was attribute support for time-series widgets not implemented in release 3.4?

**Comment by csann** [comment]
I'll add some additional information here in case someone else wants to use attributes in time series chart widgets...

In my custom widget I had to add the following to my widget javascript cod:

self.typeParameters = function() {
    return {hasAdditionalLatestDataKeys: true};
}

I placed it after my 
     self.onDataUpdated = function() {
     ...
     }
function, but it could be put anywhere in the javascript file. 

Now, when I place my custom widget on a dashboard and edit the datasources there is a new field (Latest data key) and I can add my attributes to it. Yippee!

Now I just have to figure out how to modify the custom code for the widget to access those new datasources.
